cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java

```diff
@@ -104,20 +105,7 @@ public static void setOutputPartitionMetadata(Map<String, String> arguments, Map
    * @param filter The partition filter.
    */
   public static void setInputPartitionFilter(Map<String, String> arguments, PartitionFilter filter) {
-    for (Map.Entry<String, PartitionFilter.Condition<? extends Comparable>> entry : filter.getConditions().entrySet()) {
```

Comment:
should we support the old style for backwards compatibility? If someone is passing in runtime arguments from outside instead of creating them programmatically? 

In that case, accept the old form and log a warning that this will be discontinued in the future?